moveos/moveos-verifier/src/metadata.rs

```diff
@@ -230,6 +231,65 @@ impl<'a> ExtendedChecker<'a> {
     }
 }
 
+// ----------------------------------------------------------------------------------
+// Entry Function
+
+impl<'a> ExtendedChecker<'a> {
+    fn check_entry_functions(&mut self, module: &ModuleEnv) {
+        for ref fun in module.get_functions() {
+            if !fun.is_entry() {
+                continue;
+            }
+
+            let arg_tys = &fun.get_parameter_types();
+            for ty in arg_tys {
+                self.check_transaction_input_type(&fun.get_loc(), ty);
+            }
+
+            if fun.get_return_count() > 0 {
+                self.env
+                    .error(&fun.get_loc(), "entry function cannot return values")
+            }
+        }
+    }
+
+    fn check_transaction_input_type(&self, loc: &Loc, ty: &Type) {
+        use Type::*;
+        match ty {
+            Primitive(_) | TypeParameter(_) => {
+                // Any primitive type allowed, any parameter expected to instantiate with primitive
+            }
+            Vector(ety) => {
+                // Vectors are allowed if element type is allowed
+                self.check_transaction_input_type(loc, ety)
+            }
+            Struct(mid, sid, _) if self.is_allowed_input_struct(mid.qualified(*sid)) => {
+                // Specific struct types are allowed
+            }
+            _ => {
+                // Everything else is disallowed.
```

Comment:
the `Type::Signer` and `Type::Reference(Type::Signer)` also need to support

Reply:
Fixed. Type::Signer was already a supported type in PrimitiveType, and Type::Reference(Type::Signer) type has been added.